scripts/cbmc_json_parser.py

```diff
@@ -0,0 +1,409 @@
+def is_json(myjson):
+    try:
+        json_file = json.loads(myjson)
+    except ValueError:
+        return False, None
+    return True, json_file
+
+def print_json_file(input_json_file):
+    """
+    Function for printing temporary json objects for debugging purposes
+    """
+    with open(input_json_file) as infile:
+        data = json.load(infile)
+        print(data)
+    return
+
+def restructure_json_file(log_file):
+    """
+    CBMC's response blob is difficult to parse by itself, so it needs to be re-fit into
+    another temporary dictionary structure and
+
+    Parameters -
+        log_file - log_file is the initial cbmc blob object which even in JSON format,
+                    is difficult to parse and in some cases, is not in JSON format.
+
+    Returns -
+        temp_dict - The restructured dictionary which is easy to parse
+
+    """
+
+    # The restructured dictionary which is easy to parse
+    temp_dict = {}
+    try:
+        with open(log_file) as infile:
+            data = json.load(infile)
+            temp_dict["response"] = data
+    except FileNotFoundError:
+        print("CBMC File not found")
+        return {}
+    except JSONDecodeError:
+        # print out specific information about the extra data (like the lines before it)
+        print("JSON File not being parsed")
+        return {}
+
+    # TODO: Check if passing dictionaries around is optimal or if there are alternate ways to
+    # restructure and parse cbmc blob.
+    return temp_dict
+
+def extract_solver_information(json_object):
+    """
+    Get Information about the Solver from the Json file written
+    """
+
+    # Solver information is all the fields in the json object which are not related to the results
+    solver_information = []
+    properties = []
+
+    try:
+        responses = json_object["response"]
+    except BaseException:
+        raise Exception("No Response Object from CBMC")
+
+    # Check for the objects which are not related to the result object
+    for response_object in responses:
+        if GlobalMessages.RESULT not in response_object.keys():
+            solver_information.append(response_object)
+        elif GlobalMessages.RESULT in response_object.keys():
+            properties = response_object["result"]
+        else:
+            raise KeyError
+
+    return properties, solver_information
+
+def construct_solver_information_message(solver_information):
+    """
+    From the extracted information, construct a message and append to the final Output
+
+    Sample Output -
+        CBMC 5.36.0 (cbmc-5.36.0)
+        CBMC version 5.36.0 (cbmc-5.36.0) 64-bit x86_64 linux
+        Reading GOTO program from file
+        ...
+    """
+
+    solver_information_message = ""
+    for message_object in solver_information:
+        # 'Program' and 'messageText' fields give us the information about the solver
+        try:
+            if GlobalMessages.PROGRAM in message_object.keys():
```

Comment:
Before each case, you should try to give 1-2 short abbreviated examples of what this case is intended to handle.